includes/class-database-cache.php

```diff
@@ -13,11 +13,22 @@
  * A class for caching data as an option in the database.
  */
 class Database_Cache {
-	const ACCOUNT_KEY                = 'wcpay_account_data';
-	const ONBOARDING_FIELDS_DATA_KEY = 'wcpay_onboarding_fields_data';
-	const BUSINESS_TYPES_KEY         = 'wcpay_business_types_data';
-	const CURRENCIES_KEY             = 'wcpay_multi_currency_cached_currencies';
-	const CUSTOMER_CURRENCIES_KEY    = 'wcpay_multi_currency_customer_currencies';
+	const ACCOUNT_KEY                 = 'wcpay_account_data';
+	const ONBOARDING_FIELDS_DATA_KEY  = 'wcpay_onboarding_fields_data';
+	const BUSINESS_TYPES_KEY          = 'wcpay_business_types_data';
+	const CURRENCIES_KEY              = 'wcpay_multi_currency_cached_currencies';
+	const CUSTOMER_CURRENCIES_KEY     = 'wcpay_multi_currency_customer_currencies';
+	const PAYMENT_PROCESS_FACTORS_KEY = 'wcpay_payment_process_factors';
+
+	/**
+	 * Refresh during AJAX calls is avoided, but white-listing
+	 * a key here will allow the refresh to happen.
+	 *
+	 * @var string[]
+	 */
+	const AJAX_ALLOWED_KEYS = [
+		self::PAYMENT_PROCESS_FACTORS_KEY,
+	];
```

Comment:
I saw the explanation of this new constant in the PR description, but I am still not clear about its purpose? 

I assume you want to make it to retrieve factors from the server ASAP in case something wrong. If so, I don't think this trade-off is worth and helps much in that case. Instead, may we build up a server webhook to trigger the fetching factors? That's more efficient in managing crisis, IMO.

Reply:
Yeah, this was not about doing it ASAP. There is still a 2-hour TTL for the cache, but without this change, it would never get populated. Marking this as resolved.

Reply:
>  There is still a 2-hour TTL for the cache, but without this change, it would never get populated.

Make sense, thanks for the explanation.

> Nothing to do with webhooks.

Yeah, I understand the code around here is not about webhooks. Instead, my idea is that in case of something going wrong urgently, we can send a webhook event like `payment_process_factors.updated`, which is similar to `account.updated`, and force update payment factors cache: 

https://github.com/Automattic/woocommerce-payments/blob/8c6af100520b8d0c90fde15b2e34e33c41f50a69/includes/class-wc-payments-webhook-processing-service.php#L170-L171

Reply:
To be fair, mass-sending webhooks to all sites with a given version will probably not be much quicker than just waiting for the cache to expire.

I'm open to creating another issue, and discussing this there, but I think it's out of the planned scope for this router. FWIW, I'd prefer a universal hook, which accepts DB cache keys, and deletes all of them directly, instead of a specific one for factors.

Reply:
yeah, no need to do that here, creating issue for consideration is enough. Originally, I did not understand the purpose of `AJAX_ALLOWED_KEYS`.